diff --color -uNr a/src/http/ngx_http_request.c b/src/http/ngx_http_request.c
--- a/src/http/ngx_http_request.c	2021-04-09 18:22:29.419033226 +0800
+++ b/src/http/ngx_http_request.c	2021-04-09 18:24:32.080095491 +0800
@@ -361,8 +361,7 @@
 
         /* We already have a UDP packet in the connection buffer, so we don't
          * need to wait for another read event to kick-off the handshake. */
-        cscf = ngx_http_get_module_srv_conf(hc->conf_ctx, ngx_http_core_module);
-        ngx_add_timer(rev, cscf->client_header_timeout);
+        ngx_add_timer(rev, c->listening->post_accept_timeout);
         ngx_http_quic_handshake(rev);
         return;
     }
@@ -1103,7 +1102,6 @@
     ngx_http_connection_t     *hc;
     ngx_http_v3_srv_conf_t    *qscf;
     ngx_http_ssl_srv_conf_t   *sscf;
-    ngx_http_core_srv_conf_t  *cscf;
 
     c = rev->data;
     hc = c->data;
@@ -1144,8 +1142,7 @@
     if (rc == NGX_AGAIN) {
 
         if (!rev->timer_set) {
-            cscf = ngx_http_get_module_srv_conf(hc->conf_ctx, ngx_http_core_module);
-            ngx_add_timer(rev, cscf->client_header_timeout);
+            ngx_add_timer(rev, c->listening->post_accept_timeout);
         }
 
         c->ssl->handler = ngx_http_ssl_handshake_handler;
