diff -uNr a/lib/easy.c b/lib/easy.c
--- a/lib/easy.c	2020-03-09 23:31:01.000000000 +0800
+++ b/lib/easy.c	2020-03-13 21:36:15.845461168 +0800
@@ -883,17 +883,6 @@
                              data->state.resolver))
     goto fail;
 
-#ifdef USE_ARES
-  if(Curl_set_dns_servers(outcurl, data->set.str[STRING_DNS_SERVERS]))
-    goto fail;
-  if(Curl_set_dns_interface(outcurl, data->set.str[STRING_DNS_INTERFACE]))
-    goto fail;
-  if(Curl_set_dns_local_ip4(outcurl, data->set.str[STRING_DNS_LOCAL_IP4]))
-    goto fail;
-  if(Curl_set_dns_local_ip6(outcurl, data->set.str[STRING_DNS_LOCAL_IP6]))
-    goto fail;
-#endif /* USE_ARES */
-
   Curl_convert_setup(outcurl);
 
   Curl_initinfo(outcurl);
diff -uNr a/lib/setopt.c b/lib/setopt.c
--- a/lib/setopt.c	2020-03-09 23:31:01.000000000 +0800
+++ b/lib/setopt.c	2020-03-13 21:39:23.950423488 +0800
@@ -2583,32 +2583,16 @@
 #endif
 #ifdef USE_ARES
   case CURLOPT_DNS_SERVERS:
-    result = Curl_setstropt(&data->set.str[STRING_DNS_SERVERS],
-                            va_arg(param, char *));
-    if(result)
-      return result;
-    result = Curl_set_dns_servers(data, data->set.str[STRING_DNS_SERVERS]);
+    result = Curl_set_dns_servers(data, va_arg(param, char *));
     break;
   case CURLOPT_DNS_INTERFACE:
-    result = Curl_setstropt(&data->set.str[STRING_DNS_INTERFACE],
-                            va_arg(param, char *));
-    if(result)
-      return result;
-    result = Curl_set_dns_interface(data, data->set.str[STRING_DNS_INTERFACE]);
+    result = Curl_set_dns_interface(data, va_arg(param, char *));
     break;
   case CURLOPT_DNS_LOCAL_IP4:
-    result = Curl_setstropt(&data->set.str[STRING_DNS_LOCAL_IP4],
-                            va_arg(param, char *));
-    if(result)
-      return result;
-    result = Curl_set_dns_local_ip4(data, data->set.str[STRING_DNS_LOCAL_IP4]);
+    result = Curl_set_dns_local_ip4(data, va_arg(param, char *));
     break;
   case CURLOPT_DNS_LOCAL_IP6:
-    result = Curl_setstropt(&data->set.str[STRING_DNS_LOCAL_IP6],
-                            va_arg(param, char *));
-    if(result)
-      return result;
-    result = Curl_set_dns_local_ip6(data, data->set.str[STRING_DNS_LOCAL_IP6]);
+    result = Curl_set_dns_local_ip6(data, va_arg(param, char *));
     break;
 #endif
   case CURLOPT_TCP_KEEPALIVE:
diff -uNr a/lib/urldata.h b/lib/urldata.h
--- a/lib/urldata.h	2020-03-11 06:54:05.000000000 +0800
+++ b/lib/urldata.h	2020-03-13 21:40:41.210076556 +0800
@@ -1577,11 +1577,6 @@
 
   STRING_TEMP_URL,              /* temp URL storage for proxy use */
 
-  STRING_DNS_SERVERS,
-  STRING_DNS_INTERFACE,
-  STRING_DNS_LOCAL_IP4,
-  STRING_DNS_LOCAL_IP6,
-
   /* -- end of zero-terminated strings -- */
 
   STRING_LASTZEROTERMINATED,
